app/controllers/application_controller.rb

```diff
@@ -2,4 +2,40 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+  helper_method :current_robot
+  before_action :current_order
+  before_action :find_categories
+  before_action :find_robots
+
+  def find_categories
+    @categories = Category.all
+  end
+
+  def find_robots
+    @robots = Robot.all
+  end
+
+  def current_order
+    return @current_order if !@current_order.nil?
+    if session[:order_id]
+      @current_order = Order.find(session[:order_id])
+    else
+      @current_order = Order.create(status: "pending")
```

Comment:
Maybe you could set up a default status value of "pending" with the model and/or database rather than setting it here in the controller on creation.